File: m2cgen/interpreters/rust/sigmoid.rs
fn sigmoid(x: f64) -> f64 {
    if x < 0.0_f64 {
        let z: f64 = x.exp();
        return z / (1.0_f64 + z);
    }
    return 1.0_f64 / (1.0_f64 + (-x).exp());
}

File: m2cgen/interpreters/fortran/sigmoid.f90
function sigmoid(x) result(res)
    implicit none
    double precision, intent(in) :: x
    double precision :: z

    if (x < 0.0d0) then
        z = exp(x)
        res = z / (1.0d0 + z)
    else
        res = 1.0d0 / (1.0d0 + exp(-x))
    end if

end function sigmoid
